### ISSUE TITLE
Alarm state stuck in "Pending"

### ISSUE THREAD

**Issue by davidmeixner** [asishrs/smartthings-ringalarmv2#43]
Most of the time I can arm/disarm my alarm via the app, but about 25% of the time, it gets stuck in a "Pending Home" state. It stays in that state if I try to trigger the "home" command again. If I send a "disarmed" command and then a "home" command, it generally then works to arm the alarm. I started seeing this behavior about a month ago, and continue to see it after the latest app update.

This behavior seems random, so I don't have a good method to reproduce it, and haven't been able to capture a log. Is there a way to at least force a refresh of the state so it does not stay in "Pending"? I usually automate the alarm via Webcore pistons, and if I could detect that the alarm is stuck in a pending state and force a refresh, that could serve as a workaround.

**Comment by SupaTWC** [comment]
Having the exact same problem except mine is stuck at pending off.
Please let us know which command and parameters to use to refresh the status in webcore.

**Comment by asishrs** [comment]
I have a new build with extra logging https://github.com/asishrs/smartthings/blob/master/RELEASE.md#release-v340 

Once you update to this version get me below to troubleshoot this issue.
- Get all device logs by tapping Ring Alarm Manager > Advanced Settings > Debug Logs. After you do this, you can check the logs on AWS Cloud Watch.
- Ring Alarm Manager logs from ST IDE for the devices call.
- Ring Alarm Manager logs from ST IDE for the failed status call.
- Ring Alarm Device handler logs from ST IDE for the failed status call.

**Comment by davidmeixner** [comment]
When the failed status happens, I do not see any log messages for the Alarm Manager.

I happened to have the live logging open during the failure, and have attached the Device handler log showing a triggerHome() event at 9:24:33 AM. I also attached the Alarm Manager log showing no new logs at that time. The mos recent Alarm Manager event was a status call 10 seconds earlier at 9:24:23 AM.

Also, in the AWS Cloud Watch, I only see logs for the Alarm Manager, but not the Device handler.

Let me know if there is any other information I can provide, and thank you for looking at this.

![RingAlarm](https://user-images.githubusercontent.com/15931836/79996755-cd1dfc80-8486-11ea-833f-e1e82a17fa50.PNG)
![RingAlarmManager](https://user-images.githubusercontent.com/15931836/79996756-cd1dfc80-8486-11ea-8aa0-a59a20c7e282.PNG)

**Comment by asishrs** [comment]
>When the failed status happens, I do not see any log messages for the Alarm Manager.

That's strange, is this happening frequently?

> Also, in the AWS Cloud Watch, I only see logs for the Alarm Manager, but not the Device handler.
Cloud Watch logs are for server-side code deployed in AWS lambda. It will not have logs from ST device handler or app.

**Comment by chickok77** [comment]
Mine is displaying a similar behavior in that it is always "Pending Off"

**Comment by davidmeixner** [comment]
Yes, every time this issue occurs there are no logs in the Alarm Manager. Otherwise I see logs there.

I think I have a better understanding of the problem. This seems to happens when I manually disarm the alarm from the Ring keypad. In this case, the SmartThings app still shows the alarm as armed. Then when I try to arm the alarm from WebCore (or from the app), the status of the alarm just changes to "Pending Home" and stays there. It seems that the status in the app doesn't sync with the system if it's changed from the Ring keypad.

Steps to reproduce:
1. Arm the alarm from the app
2. Disarm the alarm from the keypad
3. Arm the alarm from the app <- at this point, the app should just get stuck in the "Pending Home" state

Should the status in the app update if the alarm is armed/disarmed from the keypad? I have the polling interval set to 1 minute, but never see the status change in the app.

**Comment by esquimols6** [comment]
any update on this bug? I also have trouble getting live status of my devices on smartthings.

**Comment by asishrs** [comment]
The app is broken with the new ST app. I am not getting time to update
this.
